Analytics/Analytics/Plugins/LifecycleTrackingPlugin.swift

```diff
@@ -39,26 +41,26 @@ final class LifecycleTrackingPlugin: Plugin {
 // MARK: - LifecycleEventListener
 
 extension LifecycleTrackingPlugin: LifecycleEventListener {
+    
     func onBecomeActive() {
-        var properties: [String: Any] = [:]
-        if isFirstLaunch {
-            properties["version"] = appVersion?.currentVersionName
+        if didAppEnterBackground {
+            didAppEnterBackground = false
+            self.trackAppOpenedEvent()
         }
-        properties["from_background"] = !isFirstLaunch
-        isFirstLaunch = false
-        self.analytics?.track(name: LifecycleEvent.applicationOpened.rawValue, properties: properties)
     }
     
     func onBackground() {
+        didAppEnterBackground = true
         self.analytics?.track(name: LifecycleEvent.applicationBackgrounded.rawValue)
     }
 }
 
 // MARK: - Installation Events
 
 extension LifecycleTrackingPlugin {
-    func trackAppInstallAndUpdateEvents() {
-        guard let appVersion else { return }
+    func trackAppInstallOrUpdateEvents() {
+        guard let appVersion, isFirstLaunch else { return }
```

Comment:
I think this `isFirstLaunch` check is not needed; we can simplify it further:

```suggestion
        guard let appVersion else { return }
```